command/agent/alloc_endpoint.go

```diff
@@ -105,43 +106,70 @@ func (s *HTTPServer) ClientAllocRequest(resp http.ResponseWriter, req *http.Requ
 }
 
 func (s *HTTPServer) ClientGCRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-	if s.agent.client == nil {
-		return nil, clientNotRunning
-	}
-
-	var secret string
-	s.parseToken(req, &secret)
-
-	// Check node write permissions
-	if aclObj, err := s.agent.Client().ResolveToken(secret); err != nil {
```

Comment:
Should checking for the ACL token stay? If there is a strategy for where ACLs are/are not checked, either document it or add a comment so it is super clear.

Reply:
They should always be on the RPC which implements it not the HTTP package. The only reason these endpoints deviated was there was no RPC implementation before